### ISSUE TITLE
Intiface Central's server does not run on Steam Deck due to old version of openssl

### ISSUE THREAD

**Issue by notsafeforbread** [intiface/intiface-central#58]
Intiface Central expects openssl 3.0.x, but the current version of SteamOS still comes with `openssl 1.1.1.q-1`. Attempting to start the server fails, with no messages in the log, but if run through a terminal, the terminal window will display this:

```
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Invalid argument(s): Failed to load dynamic library 'lib/libintiface_engine_flutter_bridge.so': libssl.so.3: cannot open shared object file: No such file or directory
#0      _open (dart:ffi-patch/ffi_dynamic_library_patch.dart:12)
#1      new DynamicLibrary.open (dart:ffi-patch/ffi_dynamic_library_patch.dart:23)
#2      api (package:intiface_central/ffi.dart:23)
#3      LibraryEngineProvider.start (package:intiface_central/engine/library_engine_provider.dart)
<asynchronous suspension>
#4      EngineRepository.start (package:intiface_central/engine/engine_repository.dart:52)
<asynchronous suspension>
#5      new EngineControlBloc.<anonymous closure> (package:intiface_central/engine/engine_control_bloc.dart:76)
<asynchronous suspension>
#6      Bloc.on.<anonymous closure>.handleEvent (package:bloc/src/bloc.dart:226)
<asynchronous suspension>
```

Manually copying `libssl.so.3` and `libcrypto.so.3` onto the Steam Deck and adjusting `LD_LIBRARY_PATH` allows the Steam Deck to run the server, however.

**Comment by aphrodisihack** [comment]
I guess you tried with version 2.2. Have you tried the latest version, 2.3.0? Because I actually get the opposite problem when running that one:

```
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Invalid argument(s): Failed to load dynamic library 'lib/libintiface_engine_flutter_bridge.so': libssl.so.1.1: cannot open shared object file: No such file or directory
#0      _open (dart:ffi-patch/ffi_dynamic_library_patch.dart:11)
#1      new DynamicLibrary.open (dart:ffi-patch/ffi_dynamic_library_patch.dart:22)
#2      api (package:intiface_central/ffi.dart:23)
#3      LibraryEngineProvider.start (package:intiface_central/engine/library_engine_provider.dart)
<asynchronous suspension>
#4      EngineRepository.start (package:intiface_central/engine/engine_repository.dart:58)
<asynchronous suspension>
#5      new EngineControlBloc.<anonymous closure> (package:intiface_central/engine/engine_control_bloc.dart:93)
<asynchronous suspension>
#6      Bloc.on.<anonymous closure>.handleEvent (package:bloc/src/bloc.dart:226)
<asynchronous suspension>
```

Downgrading to 2.2.0 works for me.

**Comment by qdot** [comment]
Ugh.

The moving target here is Flutter's SSL bindings, which apparently changed between 3.3 and 3.7. I... need to figure out a better way to handle this on steam deck. Hoping the Flatpak solution works out.

**Comment by qdot** [comment]
Fixed in Desktop v2.3.0, via the Flutter 3.7 update I think? Reopen if this is still an issue.
